conans/model/conan_file.py

```diff
@@ -10,21 +10,60 @@
 from conans.paths import RUN_LOG_NAME
 
 
-def create_options(conanfile):
+class DynamicOptionsManager(object):
+    """
+    Class to allow the recipe creator to add options and standard options, using even settings
+    to condition the options
+    """
+
+    def __init__(self, options, settings):
+        self.options = options
+        self.settings = settings
+
+    def add_option(self, *args, **kwargs):
+        return self.options.add_option(*args, **kwargs)
+
+    def add_shared(self, default=False):
+        self.options.add_option("shared", [True, False], default)
+
+    def add_cppstd(self, default=None):
```

Comment:
This ``self.add_cppstd()`` default to 11 seems a bit weird to me.

Reply:
Yes, better to keep it explicit and not optional.

Reply:
Mhh, maybe it should be defaulted to the compiler's version default?